Formula/f/fake-gcs-server.rb

```diff
@@ -0,0 +1,17 @@
+class FakeGcsServer < Formula
+  desc "Google Cloud Storage emulator & testing library"
+  homepage "https://pkg.go.dev/github.com/fsouza/fake-gcs-server/fakestorage?tab=doc"
+  url "https://github.com/fsouza/fake-gcs-server/archive/refs/tags/v1.50.2.tar.gz"
+  sha256 "1607904cb3eb178575c0fdcb6b76f31f0fab21205a464c478625c90e28cf824b"
+  license "BSD-2-Clause"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
+  end
+
+  test do
+    assert_match "Usage of fake-gcs-server", shell_output("#{bin}/fake-gcs-server -h 2>&1").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate teupdate-ktlint.rb-1729781979ath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.